.github/workflows/generate-dependencies.js

```diff
@@ -0,0 +1,134 @@
+const fs = require('fs');
+const path = require('path');
+
+const ignoredFolders = ['.github', '.gradle', '.idea', '.git', 'node_modules', 'lib', 'src', 'tests'];
+const generatedFileName = 'provenance.json';
+const projectName = 'vcd-ext-sdk/';
+
+async function generateFileHelper(currentDir, components) {
```

Comment:
Can we avoid function names such as Helper, Util, Common etc. ? If the name of the function cannot be descriptive enough, let's add a code doc above.